src/libpmem2/mover.c

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2021, Intel Corporation */
+
+/*
+ * mover.c -- default pmem2 data mover
+ */
+
+#include "libpmem2.h"
+#include "mover.h"
+#include "map.h"
+#include "util.h"
+#include <stdlib.h>
+#include <unistd.h>
+
+static void
+mover_memcpy(void *arg, struct future_context *context)
+{
+	struct vdm_memcpy_data *data = future_context_get_data(context);
+	struct vdm_memcpy_output *output = future_context_get_output(context);
+	struct vdm_descriptor *dsc = (struct vdm_descriptor *)arg;
+	struct pmem2_map *map = dsc->vdm_data;
+
+	output->dest = pmem2_get_memcpy_fn(map)(data->dest, data->src,
+		data->n, PMEM2_F_MEM_NONTEMPORAL);
+	data->vdm_cb(context);
+}
+
+struct vdm *
+mover_new(struct pmem2_map *map)
+{
+	struct vdm_descriptor *desc = malloc(sizeof(*desc));
```

Comment:
Please update miniasync.
We removed the `data` field from descriptors, so now you can create an entirely static vdm descriptor. So this function will change.